controllers/actions.summerwind.net/horizontal_runner_autoscaler_webhook.go

```diff
@@ -681,78 +681,80 @@ func (autoscaler *HorizontalRunnerAutoscalerGitHubWebhook) SetupWithManager(mgr
 
 	autoscaler.Recorder = mgr.GetEventRecorderFor(name)
 
-	if err := mgr.GetFieldIndexer().IndexField(context.TODO(), &v1alpha1.HorizontalRunnerAutoscaler{}, scaleTargetKey, func(rawObj client.Object) []string {
-		hra := rawObj.(*v1alpha1.HorizontalRunnerAutoscaler)
+	if err := mgr.GetFieldIndexer().IndexField(context.TODO(), &v1alpha1.HorizontalRunnerAutoscaler{}, scaleTargetKey, autoscaler.indexer); err != nil {
```

Comment:
The last parameter of IndexField is `IndexerFunc`, which is now extracted out of this function, into a new function of the `*HorizontalRunnerAutoscalerGitHubWebhook` so that we can use the indexer func outside of this function.

`controllers/actions.summerwind.net/horizontal_runner_autoscaler_webhook_test.go` is the only place we use the indexer func outside of this func, and the indexer func is used to make the test passing after the controller-runtime upgrade.